lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -765,7 +834,41 @@ namespace Js
         CleanupBeforeExecution();
 
         Arguments outArgs(CallInfo(CallFlags_Value, 0), nullptr);
-        return rootFunction->CallRootFunction(outArgs, scriptContext, true);
+        
+        Var ret = nullptr;
+        try
+        {
+            AUTO_NESTED_HANDLED_EXCEPTION_TYPE((ExceptionType)(ExceptionType_OutOfMemory | ExceptionType_JavascriptException));
+            ENTER_SCRIPT_IF(scriptContext, true, false, false, !scriptContext->GetThreadContext()->IsScriptActive(),
+            {
+                ret = rootFunction->CallRootFunction(outArgs, scriptContext, true);
+            });
+        }
+        catch (const Js::JavascriptException& err)
+        {
+            Var errorObject = err.GetAndClear()->GetThrownObject(scriptContext);
+            AssertMsg(errorObject != nullptr && Js::JavascriptError::Is(errorObject), "ModuleEvaluation: Unhandled exception thrown by calling root function");
+            if (errorObject != nullptr && Js::JavascriptError::Is(errorObject))
+            {
+                ResolveOrRejectDynamicImportPromise(false, errorObject, scriptContext);
+                return scriptContext->GetLibrary()->GetUndefined();
+            }
+            else
+            {
+                Throw::InternalError();
+            }
+        }
+        catch (Js::OutOfMemoryException)
+        {
+            Js::JavascriptError * error = scriptContext->GetLibrary()->CreateError();
+            JavascriptError::SetErrorMessageProperties(error, E_OUTOFMEMORY, this->GetSpecifierSz(), scriptContext);
+            ResolveOrRejectDynamicImportPromise(false, error, scriptContext);
+            return scriptContext->GetLibrary()->GetUndefined();
```

Comment:
Only in the dynamic import case, we reject the promise with error object and return undefined.
In the static case, we should throw the error object. Making the change.

---
In reply to: [115383826](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115383826) [](ancestors = 115383826)